src/main/java/nl/novi/techItEasy/services/RemoteControllerService.java

```diff
@@ -14,14 +20,68 @@ public RemoteControllerService(RemoteControllerRepository repos) {
     }
 
     public Long createRemoteController(RemoteControllerDto rDto) {
-        RemoteController remote = new RemoteController();
-        remote.setCompatibleWith(rDto.compatibleWith);
-        remote.setBatteryType(rDto.batteryType);
-        remote.setName(rDto.name);
-        remote.setBrand(rDto.brand);
-        remote.setPrice(rDto.price);
-        remote.setOriginalStock(rDto.originalStock);
+        RemoteController remote = dtoToRemoteController(rDto);
         repos.save(remote);
         return rDto.getId();
     }
+
+    public List<RemoteControllerDto> getRemotes() {
+        List<RemoteController> rcList = repos.findAll();
+        List<RemoteControllerDto> rcDtoList = new ArrayList<>();
+        for (RemoteController rc : rcList) {
+            RemoteControllerDto rcDto = remoteControllerToDto(rc);
+            rcDtoList.add(rcDto);
+        }
+        return rcDtoList;
+    }
+
+    public RemoteControllerDto getRemotesById(Long id) {
+        Optional<RemoteController> r = repos.findById(id);
+        if (r.isEmpty()) {
+            throw new RecordNotFoundException("Remote controller not found");
+        } else {
+            RemoteController rc = r.get();
+            return remoteControllerToDto(rc);
+        }
+    }
+
+    public RemoteControllerDto updateRemote(Long id, RemoteControllerDto rcForUpdate) {
+        Optional<RemoteController> r = repos.findById(id);
+        if (r.isEmpty()) {
+            throw new RecordNotFoundException("Remote controller not found");
+        } else {
+            RemoteController rc = r.get();
+            RemoteController rc1 = dtoToRemoteController(rcForUpdate);
+            RemoteController returnRemote = repos.save(rc1);
```

Comment:
Je overschrijft hier de bestaande RemoteController met de update data.  Dit is een risico, als er bijvoorbeeld null waardes in de update data zitten, dan wordt dat nu gewoon overschreven.